payment/src/test/java/org/killbill/billing/payment/logging/SpyLogger.java

```diff
@@ -0,0 +1,671 @@
+package org.killbill.billing.payment.logging;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+import org.slf4j.Marker;
+import org.slf4j.helpers.FormattingTuple;
+import org.slf4j.helpers.MessageFormatter;
+
+import com.google.common.base.Optional;
+
+public class SpyLogger implements Logger {
```

Comment:
I can't extend it, because SimpleLogger has just a constructor that can be only called from within its package.